src/sync/sync-utils.ts

```diff
@@ -65,25 +51,53 @@ export const findOrStartSync = async (
 	}, 0, logger);
 };
 
-type SubscriptionUpdateTasks = {
-	totalNumberOfRepos?: number | null;
-	repositoryCursor?: string | null;
-	repositoryStatus?: string | null;
-}
+const getCommitsFromDates = async (jiraHost: string, commitsFromDate: Date | undefined) => {
+	const mainCommitsFromDate = await getCommitSinceDate(jiraHost, NumberFlags.SYNC_MAIN_COMMIT_TIME_LIMIT, commitsFromDate?.toISOString());
+	const branchCommitsFromDate = await getCommitSinceDate(jiraHost, NumberFlags.SYNC_BRANCH_COMMIT_TIME_LIMIT, commitsFromDate?.toISOString());
+	return { mainCommitsFromDate, branchCommitsFromDate };
+};
 
-const resetTargetedTasks = async (subscription: Subscription, syncType?: SyncType, targetTasks?: TaskType[]): Promise<void> => {
-	if (!targetTasks?.length) {
-		return;
+const resetRepoTaskStatusAndCursor = async (subscription: Subscription, syncType: SyncType, targetTasks?: TaskType[]) => {
+
+	const repoSyncTasks = (targetTasks || []).filter(t => t !== "repository");
+	const targetTaskIsEmpty = (targetTasks || []).length === 0;
+	const hasSubscriptionRepositoryTask = (targetTasks || []).includes("repository");
+	const hasRepoSyncTask = repoSyncTasks.length > 0;
+
+	if (syncType == "full") {
+		//For full sync
```

Comment:
comment duplicating code

Reply:
here and everywhere :)

Reply:
I have
```
} else {
		//For partial sync
```
Because it is down the track and hard to remember it is for partial.

And this `For full sync`, is just my OCD to keep it symmetric.